lambda/fulfillment/lib/middleware/1_parse.js

```diff
@@ -95,9 +95,13 @@ function getClientType(req) {
     }
     // Try to determine which Lex client is being used based on patterns in the req - best effort attempt.
     const voiceortext = (req._preferredResponseType == 'SSML') ? "Voice" : "Text" ;
-    if (_.get(req,"_event.requestAttributes.x-amz-lex:channel-type") == "Slack") {
+
+    //for LexV1 channels -- check for x-amz-lex:channel-type requestAttribute
+    //for LexV2 channels -- check for x-amz-lex:channels:platform requestAttribute
+
+    if ((_.get(req,"_event.requestAttributes.x-amz-lex:channel-type") == "Slack") || (_.get(req,"_event.requestAttributes.x-amz-lex:channels:platform") == "Slack")) {
         return "LEX.Slack." + voiceortext ;
-    } else if (_.get(req,"_event.requestAttributes.x-amz-lex:channel-type") == "Twilio-SMS") {
+    } else if ((_.get(req,"_event.requestAttributes.x-amz-lex:channel-type") == "Twilio-SMS") || (_.get(req,"_event.requestAttributes.x-amz-lex:channels:platform") == "Twilio")) {
```

Comment:
Thanks for extending the fix to twilio also, Raj!  QQ: The fix checks for V2 platform value `Twilio` rather than `Twilio-SMS` value used in V1 - just wanted to dbl check if you'd tested/verified that this is the value used for the Twilio channel in V2?

Reply:
Tested with Twilio SMS and confirm that the channel type for Lex V2 is "Twilio". 
            "requestAttributes": {
                "x-amz-lex:channels:platform": "Twilio"
            },